commandapi-core/src/main/java/dev/jorel/commandapi/config/CommentedSection.java

```diff
@@ -0,0 +1,9 @@
+package dev.jorel.commandapi.config;
+
+import org.jetbrains.annotations.ApiStatus;
+
+import java.util.List;
+
+@ApiStatus.Internal
+record CommentedSection(List<String> comment) {
```

Comment:
Yeah, I am really not a fan of this to be honest. Maybe it would make more sense if we had more sections or even nested sections but I don't think we need this at this time.